boot_services/src/boot_services.rs

```diff
@@ -333,7 +354,7 @@ pub trait BootServices: Sized {
             old_protocol_interface_ptr = old_protocol_interface as *mut _ as *mut c_void;
             new_protocol_interface_ptr = new_protocol_interface as *mut _ as *mut c_void;
         }
-        //SAFETY: The generic Protocol ensure that the interfaces is the right type for the specified protocol.
+        //SAFETY: The generic Protocol ensure that the interfaces are the right type for the specified protocol.
         unsafe {

```

Comment:
This unsafe is not necessary because the function itself is unsafe.

Reply:
Removed the unnecessary unsafe statements